app/scenes/CollectionNew.js

```diff
@@ -38,13 +38,15 @@ class CollectionNew extends React.Component<Props> {
   handleSubmit = async (ev: SyntheticEvent<>) => {
     ev.preventDefault();
     this.isSaving = true;
+    const { collections } = this.props;
     const collection = new Collection(
       {
         name: this.name,
         sharing: this.sharing,
         icon: this.icon,
         color: this.color,
         private: this.private,
+        index: collections.indexForNew,
```

Comment:
What do you think about moving this to the server? After all the client might not be aware of all the collections… which would result in the new collection potentially being inserted part way down the list.

I think _allowing_ `index` in the create is okay from an API pov though

Reply:
Let's move this to the server. If the index is `undefined` in the request payload we will insert the collection at the bottom of the collections (this is will be the default behaviour for the app). 

If the index is specified we will also need to check if the index characters are between the ASCII characters that we allow.

I was wondering, what If the index in the req is already taken? Trello API doesn't specify its behaviour for this edge case.

Reply:
> I was wondering, what If the index in the req is already taken?

I think this is where moving the index de-duplication out of the `collections.list` method comes in. If we have a de-dupe check as part of create and move then we can just adjust the item downward by one if that ever happens.